### DISCUSSION TITLE
why does waiting for  server response take too much time?

### DISCUSSION THREAD

**Discussion by mydu27** [fastapi/fastapi#11384] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typing
import orjson
from fastapi import APIRouter, Request
from starlette.responses import JSONResponse
from app.common.tools.dicom_web import default

router = APIRouter()

class JSONResponse(JSONResponse):

    def render(self, content: typing.Any) -> bytes:
        return orjson.dumps(content, default=default)


def ApiResponse(code=200, msg=None, data=None, paging=None) -> JSONResponse:
    content = {'code': code, 'msg': msg}
    if data is not None:
        content['data'] = data
    if paging is not None:
        content['paging'] = paging
    return JSONResponse(status_code=200, content=content)

@router.post("/file")
async def upload_dataset_file(*,
                              request: Request) -> ApiResponse:
    """
    upload zip file
    """
    import time
    import aiofiles

    start = time.time()
    target_folder = AsyncPath(settings.FILE_UPLOAD_PATH)
    if not await target_folder.exists():
        await target_folder.mkdir(exist_ok=True, parents=True)
    target_file_path = target_folder / (get_uuid() + '.zip')
    async with aiofiles.open(target_file_path, 'wb') as f:
        async for chunk in request.stream():
            await f.write(chunk)
    end = time.time()
    logger.info(f'it cost {end - start} seconds')
    target_file_path = str(target_file_path).replace(settings.FILE_UPLOAD_PATH,
                                                     '', 1)
    return ApiResponse(msg='successful', data=target_file_path)
```

### Description

I have a Fastapi demo and I want to use it to upload a large file. If upload a normal size file, it will be ok. But if I upload a large file( 3 GB) , write file takes about 20 seconds. Waiting for server Response takes about 50 seconds, why so late? It's ok in my windows computer. If I deployment it in my Linux Sever with Docker, it's will have this problem.
![image](https://github.com/tiangolo/fastapi/assets/37979204/0f528de8-54b5-40a8-8041-87f4a95b1bad)

### Operating System

Linux

### FastAPI Version

0.88.0

### Pydantic Version

1.10.12

### Python Version

3.9.17

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
